e2e/res/check_system/lib.rs

```diff
@@ -102,3 +106,30 @@ mod check_system {
         }
     }
 }
+
+// Haven't much thought on the API shape so far. So it isn't turned into a crate yet.
+mod js {
+    use alloc::string::String;
+    use super::*;
+
+    pub fn eval(script: &str) -> Result<String, String> {
+        use ink_env::call;
+        use pink::ConvertTo as _;
+
+        let system = pink::system::SystemRef::instance();
+        let delegate = system
+            .get_driver("JSDelegate".into())
```

Comment:
How about calling it JsEngine?

Reply:
There should be a JsEngine which is an instantiated contract id. The JSDelegate here just stores the code hash.